pkg/apiserver/user/sso/service.go

```diff
@@ -279,9 +283,6 @@ func (s *Service) discoverOIDC(issuer string) (*oidcWellKnownConfig, error) {
 		return nil, ErrDiscoverFailed.Wrap(err, "Failed to discover OIDC endpoints")
 	}
 	wellKnownConfig := resp.Result().(*oidcWellKnownConfig)
-	if wellKnownConfig.Issuer != issuer {
-		return nil, ErrDiscoverFailed.New("Issuer did not match in the OIDC provider, expect %s, got %s", issuer, wellKnownConfig.Issuer)
-	}
```

Comment:
// TODO: add an explanation about why remove it.

Reply:
Yeah, why do we remove it?

Reply:
Reasons:

1. For auth0, the `issuer` filed has `/` suffix in the `http://[org].us.auth0.com/.well-known/openid-configuration` response:

   ```json
   {
     issuer: "https://[org].us.auth0.com/",
     authorization_endpoint: "https://[org].us.auth0.com/authorize",
     token_endpoint: "https://[org].us.auth0.com/oauth/token",
     ...
   }
   ```

   For okta, the `issuer` filed has no `/` suffix in the `http://[org].okta.com/.well-known/openid-configuration` response:

   ```json
   {
     issuer: "https://dev-9991837.okta.com",
     authorization_endpoint: "https://dev-9991837.okta.com/oauth2/v1/authorize",
     token_endpoint: "https://dev-9991837.okta.com/oauth2/v1/token",
     ...
   }
   ```

   With this check, the user needs to use different rules to fill the **OIDC Discovery URL**, it confuses the user.

2. Remove this check, it still works (I have tested), even the user only fill the **OIDC Discovery URL** with `[org].us.auth0.com` or `[org].okta.com`, no `http://` or `https://` prefix and `/` suffix.

    Btw, I think this check is a little contradicting, we use an issuer value, get the correct `openid-configuration` response, but the original value is not exactly the same as the value in the response, then we tell the user input the wrong value. If the original issuer value is wrong, it should not get the correct `openid-configuration` response. So I think the check is redundant, is it right?

Reply:
I think we should keep this check (and add more checks) according to OIDC specification:

https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig

OpenID Provider Configuration Validation

If any of the validation procedures defined in this specification fail, any operations requiring the information that failed to correctly validate MUST be aborted and the information that failed to validate MUST NOT be used.

The issuer value returned MUST be identical to the Issuer URL that was directly used to retrieve the configuration information. This MUST also be identical to the iss Claim value in ID Tokens issued from this Issuer.

Reply:
Ok, let me have a look.

Reply:
Revert and add more checks.